### ISSUE TITLE
[GR-52711] Add compiler intrinsics for GraalWasm SIMD instructions

### ISSUE THREAD

**Issue by fniephaus** [oracle/graal#9767]
### TL;DR

We plan to improve the performance of GraalWasm's implementation of [the SIMD proposal](https://github.com/WebAssembly/simd).

### Details

GraalWasm implements the instructions of the SIMD proposal, which is part of the WebAssembly 2.0 draft specification. However, our current implementation does not enjoy the benefits of parallel execution, leading to suboptimal performance. Our goal is to bring GraalWasm and the Graal compiler together and make it so that WebAssembly SIMD instructions end up compiled as hardware SIMD instructions.
This should lead to substantial performance gains when running modules which make use of the SIMD proposal.
